### DISCUSSION TITLE
We love PyMDE! -- What is its future?

### DISCUSSION THREAD

**Discussion by eric79** [cvxgrp/pymde#75] [General]
Hi, @akshayka !  Thanks for creating pymde.  We love it.  We notice that it has a pre-release version number.  Do you have plans to strike and support a 1.0 version at some point?  If so, do you have a rough timeline in mind?

**Comment by akshayka** [comment]
Hi Eric! Thanks for the kind words. Out of curiosity, if you can say, what do you use PyMDE for?

I believe I haven't made any backward incompatible changes, and don't plan on making any. However, the reason the software is still pre 1.0 is that I'd just like to reserve the right to do so in case some unforeseen problem with the current API comes up.

I was thinking of releasing a 1.0 version when I become very very confident in the APIs and in the stability of the software. I do not yet have a timeline for that.

In the meantime, I guarantee that any backward compatible change will result in a version bump in minor part of the version (eg, 0.1.x to 0.2.0). I may also bump a minor version if significant new features are added. In any case, I will make sure that such a release has detailed release notes.

**Comment by ErrorUnknown88** [comment]
Hey @akshayka,
We use pymde to replace a dependency on UMAP. We use dimensionality reduction before clustering in many domains and so combine the lower dimensions with HDBSCAN. Pymde offers us GPU support that UMAP cannot without installing the cu-ml packages. For our use cases it appears that the method outperforms UMAP after clustering by giving us greater separation and coherence. So we get a speed boost as well as a better set of groups.
